From 92fa2850b5204482962740ec326a9ae938e3dd62 Mon Sep 17 00:00:00 2001
From: Hediberto Cavalcante da Silva <hediberto.cavalcantedasilva@windriver.com>
Date: Tue, 16 May 2023 14:55:15 -0300
Subject: [PATCH] csm-application-mobility: support dynamic rbac value

This patch consists of changing the rbac image value from static to dynamic
allowing the system reads the image in values.yaml and add the registry local prefix.

Signed-off-by: Hediberto Cavalcante da Silva <hediberto.cavalcantedasilva@windriver.com>
Signed-off-by: Gustavo Ornaghi Antunes <gustavo.ornaghiantunes@windriver.com>
---
 .../apps_v1_deployment_app-mobility-controller-manager.yaml    | 2 +-
 charts/csm-application-mobility/values.yaml                    | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/charts/csm-application-mobility/templates/apps_v1_deployment_app-mobility-controller-manager.yaml b/charts/csm-application-mobility/templates/apps_v1_deployment_app-mobility-controller-manager.yaml
index ec4f9dd..b353fe2 100644
--- a/charts/csm-application-mobility/templates/apps_v1_deployment_app-mobility-controller-manager.yaml
+++ b/charts/csm-application-mobility/templates/apps_v1_deployment_app-mobility-controller-manager.yaml
@@ -23,7 +23,7 @@ spec:
         - --upstream=http://127.0.0.1:8080/
         - --logtostderr=true
         - --v=10
-        image: gcr.io/kubebuilder/kube-rbac-proxy:v0.8.0
+        image: {{ .Values.rbac.image }}
         name: kube-rbac-proxy
         ports:
         - containerPort: 8443
diff --git a/charts/csm-application-mobility/values.yaml b/charts/csm-application-mobility/values.yaml
index 5e091ed..40be3a6 100644
--- a/charts/csm-application-mobility/values.yaml
+++ b/charts/csm-application-mobility/values.yaml
@@ -10,6 +10,9 @@ image:
 controller:
   image: dellemc/csm-application-mobility-controller:v0.3.0
 
+rbac:
+  image: gcr.io/kubebuilder/kube-rbac-proxy:v0.11.0
+
 # csm-application-mobility requires cert-manager. If cert-manager is not already present in cluster, set enabled to true to install it too.
 cert-manager:
   enabled: false
-- 
2.25.1
